app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,39 @@
+class MerchantsController < ApplicationController
+  def index
+    @merchants = Merchant.all
+  end
+
+  def show
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def new
+  end
+
+  def create
+    Merchant.create(merchant_params)
+    redirect_to '/merchants'
+  end
+
+  def edit
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def update
+    @merchant = Merchant.find(params[:id])
+    @merchant.update(merchant_params)
+    @merchant.save
```

Comment:
It's not necessary to do `@merchant.save` here. `update` will automatically save for you.